File: api/src/main/java/com/example/api/dto/BookingDto.java
package com.example.api.dto;

import com.example.api.dto.enums.BookingStatus;
import io.swagger.v3.oas.annotations.media.Schema;

import java.time.LocalDateTime;

@Schema(description = "Запрос на бронирование")
public record BookingDto(@Schema(description = "Идентификатор запроса") Long id,
                         @Schema(description = "Дата и время начала бронирования") LocalDateTime start,
                         @Schema(description = "Дата и время окончания бронирования") LocalDateTime end,
                         @Schema(description = "Предмет, который пользователь бронирует") ItemDto item,
                         @Schema(description = "Пользователь, который осуществляет бронирование") UserDto booker,
                         @Schema(description = "Статус бронирования") BookingStatus status) {
}

File: api/src/main/kotlin/com/example/api/dto/BookingDto.kt
package com.example.api.dto

import com.example.api.dto.enums.BookingStatus
import io.swagger.v3.oas.annotations.media.Schema
import java.time.LocalDateTime

@Schema(description = "Информация о запросе на бронирование")
data class BookingDto(
    @get:Schema(description = "Идентификатор запроса на бронирование") val id: Long,
    @get:Schema(description = "Дата и время начала бронирования") val start: LocalDateTime,
    @get:Schema(description = "Дата и время окончания бронирования") val end: LocalDateTime,
    @get:Schema(description = "Предмет, который пользователь бронирует") val item: ItemDto,
    @get:Schema(description = "Пользователь, который осуществляет бронирование") val booker: UserDto,
    @get:Schema(description = "Статус бронирования") val status: BookingStatus
)
